utils/connectWallet.ts

```diff
@@ -0,0 +1,47 @@
+import Onboard from "@web3-onboard/core";
+import safeModule from "@web3-onboard/gnosis";
+import trustModule from "@web3-onboard/trust";
+import ledgerModule from "@web3-onboard/ledger";
+import trezorModule from "@web3-onboard/trezor";
+import coinbaseModule from "@web3-onboard/coinbase";
+import walletConnectModule from "@web3-onboard/walletconnect";
+import injectedModule from "@web3-onboard/injected-wallets";
+
+const injected = injectedModule();
+
+export const onboard = Onboard({
+  wallets: [
+    injected,
+    walletConnectModule({
+      projectId: "your-project-id",
+    }),
+    coinbaseModule(),
+    ledgerModule({
+      walletConnectVersion: 2,
+      projectId: "your-project-id",
+    }),
+    trezorModule({
+      email: "test@test.com",
+      appUrl: "https://www.example.com",
```

Comment:
These should be environment variables
```suggestion
      email: process.env.NEXT_PUBLIC_SUPPORT_EMAIL,
      appUrl: process.env.NEXT_PUBLIC_APP_URL,
```